libsoftwarecontainer/src/gateway/networkgateway.cpp

```diff
@@ -347,3 +333,54 @@ bool NetworkGateway::isBridgeAvailable()
     return isSuccess(m_netlinkHost.hasAddress(addresses, AF_INET, m_gateway.c_str()));
 }
 
+std::string IPTableEntry::getChain() {
+    if ( "INCOMING" == m_type) {
+        return "INPUT";
+    } else if ("OUTGOING" == m_type) {
+        return "OUTPUT";
+    } else if ("FORWARD" == m_type) {
+        return "FORWARD";
+    }
+    return "";
```

Comment:
This is not a very nice solution. An enum would serve the purpose better imo, and could be compared with in insertRule() also.